velox/type/TimestampConversion.cpp

```diff
@@ -866,8 +1002,14 @@ fromTimestampWithTimezoneString(
     }
 
     std::string_view timeZoneName(str + pos, timezonePos - pos);
+    std::string normalizeTzName = std::string{timeZoneName};
+    if (parseMode == TimestampParseMode::kSparkCast &&
+        !normalizeSparkTimezone(normalizeTzName, resultTimestamp)) {
+      return folly::makeUnexpected(Status::UserError(
+          "Failed to normalize spark timezone value: \"{}\"", timeZoneName));
+    }
 
-    if ((timeZone = tz::locateZone(timeZoneName, false)) == nullptr) {
+    if ((timeZone = tz::locateZone(normalizeTzName, false)) == nullptr) {
       return folly::makeUnexpected(
           Status::UserError("Unknown timezone value: \"{}\"", timeZoneName));
```

Comment:
nit: use 'normalizeTzName' in the error msg.

Reply:
I deliberately used `timeZoneName` in the msg because `timeZoneName` is the original timezone string, while `normalizeTzName` has been modified. I want the user to see that the original timezone is problematic in the error msg.